BullsAndCowsTest/BullsAndCowsGameTest.cs

```diff
@@ -8,10 +10,124 @@ public class BullsAndCowsGameTest
         [Fact]
         public void Should_create_BullsAndCowsGame()
         {
-            var secretGenerator = new SecretGenerator();
-            var game = new BullsAndCowsGame(secretGenerator);
+            Mock<SecretGenerator> mockedSecretGenerator = new Mock<SecretGenerator>();
+            mockedSecretGenerator.Setup(generator => generator.GenerateSecret()).Returns("");
+
+            var game = new BullsAndCowsGame(mockedSecretGenerator.Object);
             Assert.NotNull(game);
-            Assert.True(game.CanContinue);
         }
+
+        [Fact]
+        public void Should_return_4A0B_when_guess_given_guess_number_and_secret_are_same()
+        {
+            var guessNumber = "1234";
+            var secret = "1234";
+
+            Mock<SecretGenerator> mockedSecretGenerator = new Mock<SecretGenerator>();
+            mockedSecretGenerator.Setup(generator => generator.GenerateSecret()).Returns(secret);
+
+            var game = new BullsAndCowsGame(mockedSecretGenerator.Object);
+            var result = game.Guess(guessNumber);
+            Assert.Equal("4A0B", result);
+        }
+
+        [Fact]
+        public void Should_return_2A0B_when_guess_given_guess_number_and_secret_have_same_number_in_same_positions()
```

Comment:
Test name could be 'Should_return_2A0B_when_guess_given_guess_number_and_secret_have_same_number_in_same_positions_partical', which matches test code more.